### PR TITLE
Add support for NSX-T Edge Gateway BGP IP Prefix lists

### PR DESCRIPTION
**Pull request by Didainius**
This PR continues to add support for BGP configuration (as PR #480)  for BGP and adds support for  IP Prefix Lists. These list can be consumed when defining BGP neighbors (a PR yet to be done)

It adds types `EdgeBgpIpPrefixList` and `types.EdgeBgpIpPrefixList` with functions `CreateBgpIpPrefixList`, `GetAllBgpIpPrefixLists`, `GetBgpIpPrefixListByName`, `GetBgpIpPrefixListById`, `Update` and `Delete`

**Note.** VCD 10.2 versions have a problem - they do not return a new entity ID when it is created. Function `CreateBgpIpPrefixList` handles this internally by waiting for task completion and checking if ID was returned (it is returned in 10.3). If the ID is not returned - the function will lookup the entity by Name and return it with the corresponding ID. **There is a risk that** - if more than one entity with such a name exists - it will fail to find a correct object and return an error. 

VCD Docs ->  https://docs.vmware.com/en/VMware-Cloud-Director/10.3/VMware-Cloud-Director-Service-Provider-Admin-Portal-Guide/GUID-F2A3BC91-036A-4E29-A1C9-6EAB8602562E.html

### CODE REVIEW

File: govcd/nsxt_edgegateway_bgp_ip_prefix_list.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("error creating NSX-T Edge Gateway BGP IP Prefix List: %s", err)
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("error creating NSX-T Edge Gateway BGP IP Prefix List: %s", err)
+	}
+
+	// API has problems therefore explicit manual handling is required to lookup newly created object
```
Comment (lvirbalas) [inline govcd/nsxt_edgegateway_bgp_ip_prefix_list.go:57]:
```suggestion
	// API is not consistent across different versions therefore explicit manual handling is required to lookup newly created object
```

File: govcd/nsxt_edgegateway_bgp_ip_prefix_list.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+	// VCD 10.2 -> no ID for newly created object is returned at all
+	// VCD 10.3 -> `Details` field in task contains ID of newly created object
+	// To cover all cases this code will at first look for ID in `Details` field and fall back to
+	// lookup by name if `Details` field is empty.
+	//
+	// The drawback of this is that it is possible to create duplicate records with the same name on VCDs that don't
```
Comment (lvirbalas) [inline govcd/nsxt_edgegateway_bgp_ip_prefix_list.go:63]:
```suggestion
	// The drawback of this is that it is possible to create duplicate records with the same name on VCD versions that don't
```

File: types/v56/constants.go [hunk 3]
```diff
@@ -375,6 +375,8 @@ const (
 	OpenApiEndpointSecurityTags                       = "securityTags"
 	OpenApiEndpointNsxtRouteAdvertisement             = "edgeGateways/%s/routing/advertisement"
 
+	OpenApiEndpointEdgeBgpConfigPrefixLists = "edgeGateways/%s/routing/bgp/prefixLists/" // '%s' is NSX-T Edge gateway ID
```
Comment (lvirbalas) [inline types/v56/constants.go:378]:
```suggestion
	OpenApiEndpointEdgeBgpConfigPrefixLists = "edgeGateways/%s/routing/bgp/prefixLists/" // '%s' is NSX-T Edge Gateway ID
```

**Review by lvirbalas — COMMENTED** [review]
Looks great, just some suggestions here and there.

**Review by adambarreiro — APPROVED** [review]
LGTM! I've pending to run the tests.

File: govcd/nsxt_edgegateway_bgp_ip_prefix_list.go (hunk 1 above)
Comment (Didainius) [reply govcd/nsxt_edgegateway_bgp_ip_prefix_list.go]: Thanks, this reads a lot better.

File: govcd/nsxt_edgegateway_bgp_ip_prefix_list.go [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+	// return IDs, but there is no better way for VCD versions that don't return API code
```
[the remark this answers is not in the document]
Comment (Didainius) [reply govcd/nsxt_edgegateway_bgp_ip_prefix_list.go]: This was some brain error. Adjusted.

File: types/v56/constants.go (hunk 3 above)
Comment (Didainius) [reply types/v56/constants.go]: Done.

**Review by lvirbalas — APPROVED** [review]
LGTM!

**Pull request merged**